pyiode/pyiode_model.pyx

```diff
@@ -6,45 +6,246 @@
     
+    Parameters
+    ----------
+    nb_passes: int
+        number of iterations of the reordering algo (aka triangulation)
+    pre_listname: str
+        name of the iode list where the prolog block of equations must be stored
+    inter_listname: str 
+        name of the iode list where the interdependent block of equations must be stored
+    post_listname: str 
+        name of the iode list where the epilog block of equations must be stored
+    eqs_list
+        list of equations defining the model or None for all equations in the current WS of equations
+        
+    Returns
+    -------    
+    No return value.  On error, raises an exception .
+    
+    Examples
+    --------
+    >>> iode.model_calc_scc(nb_passes=1)        # doctest: +SKIP
     
+    '''
+    if isinstance(eqs_list, list):
+        eqs_list = ','.join(eqs_list)
+    elif eqs_list is None:
+        eqs_list = ""
+    
+    if IodeModelCalcSCC(nb_passes, 
+                         cstr(pre_listname), cstr(inter_listname), cstr(post_listname),
+                         cstr(eqs_list)):
+        raise RuntimeError(f"Cannot create the model Connex Components")
+    
+
+def model_simulate_scc( sample_from: str, sample_to: str, 
+                        pre_listname: str = "_PRE", 
+                        inter_listname: str = "_INTER", 
+                        post_listname: str = "_POST",
+                        eps: float = 0.0001, 
+                        relax: float = 1.0, 
+                        maxit: int = 100, 
+                        init_values: int = KV_INIT_TM1, 
+                        debug: int = 0, 
+                        newton_eps: float = 1e-6, 
+                        newton_maxit: int = 50, 
+                        newton_debug: int = 0):
+    '''
+    Simulate a model defined by the 3 Connex Components (stored in 3 lists).
+    '''
+
+    model_simulate_save_parms(eps, relax, maxit, init_values, -1, -1)
+    
+    if IodeModelSimulateSCC(cstr(sample_from), cstr(sample_to), 
+                         cstr(pre_listname), cstr(inter_listname), cstr(post_listname),
+                         eps, relax, maxit, init_values, debug, 
+                         newton_eps, newton_maxit, newton_debug):
+        raise RuntimeError(f"model_simulate_scc() failed")
+        
+
+def model_simulate_save_parms(
+                    eps: float = 0.0001, 
+                    relax: float = 1.0, 
+                    maxit: int = 100, 
+                    init_values: int = KV_INIT_TM1, 
+                    sort_algo: int = SORT_BOTH, 
+                    nb_passes: int = 1):
+    ''' 
+        Save in the global variabels KSIM_* the simulation parameters used during the last 
+        call to model_simulate() and model_simulate_scc(). 
+        The purpose of this function is to enable later reporting
+        via the functions model_simulate_maxit(), model_simulate_eps()...
+    '''
+    global KSIM_MAXIT
```

Comment:
No, don't use global. `global` is only for very specific cases in Python. Please do not defined `global` values. 

You may simply declare all these variables OUTSIDE of the function with a default initialisation value. Also, start the name of these variables with two underscores so that the statement `from iode import *` wil NOT import those 'global' variables.

In short, do NOT use the keyword `global` enywhere in the Cython code for the iode module.
Please

Reply:
OK. If `global` is removed, I am happy.